src/MeshKernelNET/Api/MeshKernelApi.cs

```diff
@@ -234,6 +234,7 @@ public int CurvilinearGridGetData(int meshKernelId, out DisposableCurvilinearGri
         public int CurvilinearGetBoundariesAsPolygons(int meshKernelId, int lowerLeftN, int lowerLeftM, int upperRightN,  int upperRightM, out DisposableGeometryList boundaryPolygons)

```

Comment:
Is it better to use a local buffer?

Reply:
`public int CurvilinearGetBoundariesAsPolygons(int meshKernelId, int lowerLeftN, int lowerLeftM, int upperRightN,  int upperRightM, out DisposableGeometryList boundaryPolygons)
        {
            int numberOfPolygonNodes = 0; 
            boundaryPolygons = new DisposableGeometryList();
            int exitCode = MeshKernelDll.CurvilinearCountGetBoundariesAsPolygons(meshKernelId, 
                                                                                 lowerLeftN, 
                                                                                 lowerLeftM, 
                                                                                 upperRightN, 
                                                                                 upperRightM, 
                                                                                 ref numberOfPolygonNodes);
            if (exitCode != 0)
            {
                return exitCode;
            }

            // prepare the data structure for pinning
            using (var exchangePolygons = new DisposableGeometryList())
            {
                exchangePolygons.NumberOfCoordinates = numberOfPolygonNodes;
                exchangePolygons.XCoordinates = new double[numberOfPolygonNodes];
                exchangePolygons.YCoordinates = new double[numberOfPolygonNodes];
                exchangePolygons.GeometrySeparator = GetSeparator();
                exchangePolygons.InnerOuterSeparator = GetInnerOuterSeparator();

                var exchangePolygonsNative = exchangePolygons.CreateNativeObject();
                exitCode = MeshKernelDll.CurvilinearGetBoundariesAsPolygons(meshKernelId,
                                                                            lowerLeftN,
                                                                            lowerLeftM,
                                                                            upperRightN,
                                                                            upperRightM,
                                                                            ref exchangePolygonsNative);

                if (exitCode != 0)
                {
                    return exitCode;
                }

                boundaryPolygons.XCoordinates = exchangePolygonsNative.xCoordinates.CreateValueArray<double>(numberOfPolygonNodes);
                boundaryPolygons.YCoordinates = exchangePolygonsNative.yCoordinates.CreateValueArray<double>(numberOfPolygonNodes);
                boundaryPolygons.NumberOfCoordinates = exchangePolygons.NumberOfCoordinates;
                boundaryPolygons.GeometrySeparator = exchangePolygons.GeometrySeparator;
                boundaryPolygons.InnerOuterSeparator = exchangePolygons.InnerOuterSeparator;
                return exitCode;
            }
        }`

Reply:
done.